### ISSUE TITLE
No Icon

### ISSUE THREAD

**Issue by polygonnedpotato** [MoralCode/ClassClock#22]
While the app does have an icon, never in the app itself is it used. this would be a great thing to fix in the future.

**Comment by MoralCode** [comment]
Thanks for the suggestion!
To be honest I think the icon originally created just so that there was _something_ to use as a logo and/or home screen icon.

That said, if you have any ideas for new logo designs or ways that the icon can be used within the app, I would definitely be open to considering them!

**Comment by polygonnedpotato** [comment]
well, for starters, there should be a favicon and icons formatted for all devices (ex: apple iPhone, android, etc.)

here is a list of possible icon names that can be used and how they are used:
favicon.ico -- Favorites icon and icon that appears on tab
favicon-16x16.png -- alt favicon
favicon-32x32.png -- lager alt favicon
android-chrome-192x192.png -- Google chrome icon
android-chrome-256x256.png -- larger Google Chrome icon
apple-touch-icon.png -- Safari icon
mstile-150x150.png -- Microsoft Edge icon (Windows 10)
safari-pinned-tab.svg -- Safari icon (macOS X)

that should be all.

**Comment by MoralCode** [comment]
ohhh that makes more sense, I thought you meant like displaying the icon to the user inside the `<body>` of the page.

There are already 512px and 192 px icons defined both [here](https://github.com/MoralCode/ClassClock/blob/master/public/index.html#L24-L27) for iOS and in the `manifest.json` file for android as I was under the impression that these would be automatically down-scaled to the appropriate smaller sizes (and effectively replace the `.ico` files), however if this assumption is not the case then I can add that to the backlog.

Thanks for the detailed list!

**Comment by polygonnedpotato** [comment]
actually, the app icon is never displayed at all on any device. for example...
![image](https://user-images.githubusercontent.com/73083772/98176946-6904d800-1eae-11eb-994a-4e177df46d3a.png)
this is the app pinned in chrome os
![image](https://user-images.githubusercontent.com/73083772/98176996-85a11000-1eae-11eb-9b3c-8ceeff601eae.png)
and this is the app in chrome.

**Comment by polygonnedpotato** [comment]
Here is an example that can be used
`              <title>ClassClock</title>
		<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png?v=LbWArJzeO7">
		<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png?v=LbWArJzeO7">
		<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png?v=LbWArJzeO7">
		<link rel="manifest" href="/site.webmanifest?v=LbWArJzeO7">
		<link rel="mask-icon" href="/safari-pinned-tab.svg?v=LbWArJzeO7" color="#5bbad5">
		<link rel="shortcut icon" href="/favicon.ico?v=LbWArJzeO7">
		<meta name="apple-mobile-web-app-title" content="ClassClock">
		<meta name="application-name" content="ClassClock">
		<meta name="msapplication-TileColor" content="#2b5797">
		<meta name="theme-color" content="#008080">		
		<meta name="viewport" content="width=device-width, user-scalable=no">`

**Comment by MoralCode** [comment]
well that definitely wasn't supposed to happen. Thanks for the sample code!

I'm not sure when I will be able to get around to making this change though, so if you'd like to make a pull request, that would be amazing!

**Comment by polygonnedpotato** [comment]
I just made a good icon for the mstile version
![mstile-512](https://user-images.githubusercontent.com/73083772/98701936-ad6cf980-232e-11eb-8ea8-960cfa2f671b.png)

**Comment by MoralCode** [comment]
upon some further investigation, it seems like thee icons may have been configured correctly, but [UniversalRouter](https://github.com/kriasoft/universal-router) inside the app is not configured to correctly returning the appropriate assets as it tries to interperet the pages as react routes.
![Screenshot_20201201_130600](https://user-images.githubusercontent.com/17362949/100796736-f8ca7300-33d5-11eb-90ce-1668f9386f8d.png)

**Comment by MoralCode** [comment]
also probably caused by the same or similar routing issues that are causing #25

**Comment by MoralCode** [comment]
this could also be caused by whatever software is bundling the app up seemingly not including the image assets from the `/icons` directory

**Comment by polygonnedpotato** [comment]
Another reason it might not be showing up is that the favicon is not properly configured. I am going to fork this to Repl.it and run some tests to see if I can fix it. I will report back with the information I found.

**Comment by MoralCode** [comment]
does the `rel` attribute not allow multiple values like `class` does?
i thought that[ this line](https://github.com/MoralCode/ClassClock/blob/77ea42d4d225bf03bab63787c478b27de8f20b09/public/index.html#L26) added the icon with a `rel` value of both `shortcut` and `icon`, but there's also so many different things going on with this that its somewhat difficult to tell what is actually causing the problems

**Comment by polygonnedpotato** [comment]
When you create a shortcut in a browser, it relies on the favicon and uses that.

**Comment by MoralCode** [comment]
turns out that all that was needed was to move the icons to the `public` folder so that whatever web-packing software ClassClock is using (probably the one that comes with create-react-app) knows to just copy it to the built site. 

Should be fixed in fab5c5aff73f4b3b8b479a8f265e2c4e007da4bd

**Comment by polygonnedpotato** [comment]
OK I'll check it out

**Comment by MoralCode** [comment]
huh ok, To be fair i also wasnt fully testing every icon and was mostly just going off of the fact that the favicon now shows up when running the dev server as compared to the live app. I'll probably run a lighthouse test as well just to see what that says
![Screenshot_20201224_104757](https://user-images.githubusercontent.com/17362949/103103396-a1cf4c80-45d5-11eb-8a26-464125a63cec.png)

**Comment by polygonnedpotato** [comment]
I already did some and will send them to you.

**Comment by polygonnedpotato** [comment]
I JUST FOUND OUT WHAT'S GOING ON!
The icon does not show up as the icons lead to a 404 page. it is possible that the icons were not uploaded correctly and thus failing to load the icon.

**Comment by MoralCode** [comment]
Good catch. I was being dumb and put the icons in the wrong place. 

Are you testing with the `main` branch or the `development` branch? When I work locally on the `development` branch I am able to see the favicon loading correctly in my browser tab.
![Screenshot_20210120_143715](https://user-images.githubusercontent.com/17362949/105249493-0aeea500-5b2d-11eb-9aa9-ead31751da27.png)

Please let me know if there are still missing icons on the `development` branch so I can make sure they get added.

**Comment by polygonnedpotato** [comment]
i am just testing the online version. i found it out via the javascript console.

**Comment by polygonnedpotato** [comment]
it turns out that the online version relies on the `gh-pages` branch, meaning that theoretically if the file `classclockicon-192.png` was added to a folder named icons in the `gh-pages` branch, the favicon would show up.

**Comment by MoralCode** [comment]
Yep, thats pretty much what happened in commit fab5c5a, i think there was already an `icons` folder in the repository, but when webpack generates builds of the site for releasing to `gh-pages` it seemed to be skipping that folder rather than copying it to the `gh-pages` branch since it was not in the `public` folder where static assets are copied from.

after this commit, icons in `public/icons` in the repository should be copied to `/icons` in `gh-pages` and show up properly. Currently this commit is only on the development branch so it wont be publicly visible on the main ClassClock site until it is merged to `main`, however, it should be visible on the beta site at https://beta.web.classclock.app if you want to test it out
